lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -337,6 +348,16 @@ def _connect_uncached(self):
             # paramiko 2.2 introduced auth_timeout parameter
             if LooseVersion(paramiko.__version__) >= LooseVersion('2.2.0'):
                 ssh_connect_kwargs['auth_timeout'] = self._play_context.timeout
+            else:
+                display.warning('auth_timeout ignored.'
+                                ' Please upgrade to Paramiko 2.2.0 or newer.')
+
+            # paramiko 1.15 introduced banner timeout parameter
+            if LooseVersion(paramiko.__version__) >= LooseVersion('1.15.0'):
+                ssh_connect_kwargs['banner_timeout'] = self.get_option('banner_timeout')
+            else:
+                display.warning('banner_timeout ignored.'
+                                ' Please upgrade to Paramiko 1.15.0 or newer.')
```

Comment:
After seeing the warning behavior, I don't think the warnings are a good idea. I see two issues:

1) The warnings are shown on every task.
2) The warnings are shown even if the user did not set the timeouts.

Instead, I'd just document the version limitation for the `banner_timeout` setting. If we had a way to resolve both of the above issues then the warnings would probably be a good idea.

Reply:
Yes agreed, missed the point about warning on every task. Have updated the PR to reflect the paramiko version limitation in option description.